src/cppcache/include/geode/SharedBase.hpp

```diff
@@ -56,11 +57,12 @@ class CPPCACHE_EXPORT SharedBase {
 
  protected:
   inline SharedBase(bool noInit) {}
+  inline SharedBase(const SharedBase&) {}
 
   virtual ~SharedBase() {}
 
  private:
-  mutable volatile int32_t m_refCount;
+  std::atomic<int32_t> m_refCount;
```

Comment:
It didn't appear that the `mutable` attribute was necessary for operations on this value. This `SharedBase` thing is a little funky so I didn't poke too deep. The compiler did not find any issues where a `const` function or reference was making changes to this value. It may be worth more investigation.